source/code/plugins/in_npmd.rb

```diff
@@ -0,0 +1,676 @@
+class Logger
+    require 'thread'
+
+    LogDepthIncResc = 1 # Depth increase of method scope for rescue
+    LogDepthIncLoop = 2 # Depth increase of method scope for loop
+
+    def self.getMethodname(depth)
+        _depth = depth > 0 ? depth : 0
+        begin
+            caller_locations(2 + _depth, 1)[0].label
+        rescue
+            caller_locations(2 + LogDepthIncResc, 1)[0].label
+        end
+    end
+    def self.logError(msg, depth=0)
+        _methodname = getMethodname(depth)
+        _message = "[#{_methodname}]:#{msg}"
+        $log.error "[NPMD]:#{_message}"
+    end
+    def self.logInfo(msg, depth=0)
+        _methodname = getMethodname(depth)
+        $log.info "[NPMD]:[#{_methodname}]:#{msg}"
+    end
+    def self.logWarn(msg, depth=0)
+        _methodname = getMethodname(depth)
```

Comment:
If depth does not change, you can have getMethodname (depth) as glbal param . Function call will be done only once, then throughout the code just use the param value instead of calling the same function multiple times